src/modules/name/index.ts

```diff
@@ -247,22 +247,43 @@ export class Name {
   /**
    * Return a random gender.
    *
-   * @param binary Whether to return only binary gender names. Defaults to `false`.
+   * @param binary (deprecated) Whether to return only binary gender names. Defaults to `false`.
+   *
+   * @see faker.name.sex() if you would like to generate binary-gender value
    *
    * @example
    * faker.name.gender() // 'Trans*Man'
-   * faker.name.gender(true) // 'Female'
    */
   gender(binary?: boolean): string {
     if (binary) {
-      return this.faker.helpers.arrayElement(
-        this.faker.definitions.name.binary_gender
-      );
+      deprecated({
+        deprecated: 'faker.name.gender(true)',
+        proposed: 'faker.name.sex()',
+        since: '7.4',
```

Comment:
```suggestion
        since: '7.5',
```

If there are other places in this PR, please change them as well